### ISSUE TITLE
GET orders returns error when filtering to status=pending

### ISSUE THREAD

**Issue by amnotafraid** [magento/magento2#8947]
### Preconditions
I have a Magento version 2.1.3 installation.  It has many orders wth different statuses.
### Steps to reproduce
If I do a GET to this URL (replace BASE_URL with magento 2 url, send token in header), I get a list of orders as expected (this is status="canceled"):
```
BASE_URL/rest/V1/orders/?searchCriteria[filter_groups][0][filters][0][field]=status&searchCriteria[filter_groups][0][filters][0][value]=canceled&searchCriteria[filter_groups][0][filters][0][condition_type]="eq"&searchCriteria[currentPage]=1&searchCriteria[pageSize]=10
```
If I simply change this to status="pending", I get an error.  This is the URL:
```
BASE_URL/rest/V1/orders/?searchCriteria[filter_groups][0][filters][0][field]=status&searchCriteria[filter_groups][0][filters][0][value]=pending&searchCriteria[filter_groups][0][filters][0][condition_type]="eq"&searchCriteria[currentPage]=1&searchCriteria[pageSize]=10
```
### Expected result
I expect to get pending orders when I GET to the second URL.  There are 8 of them on the Magento install I am working with. 

### Actual result
When I do a GET with the second URL, I get this error:
```
{
  "message": "Notice: Array to string conversion in /srv/weldioc/html/vendor/magento/framework/Reflection/TypeCaster.php on line 34",
  "trace": "#0 /srv/weldioc/html/vendor/magento/framework/Reflection/TypeCaster.php(34): Magento\\Framework\\App\\ErrorHandler->handler(8, 'Array to string...', '/srv/weldioc/ht...', 34, Array)\n#1 /srv/weldioc/html/vendor/magento/framework/Reflection/DataObjectProcessor.php(139): Magento\\Framework\\Reflection\\TypeCaster->castValueToType(Array, 'string')\n#2 /srv/weldioc/html/vendor/magento/framework/Reflection/DataObjectProcessor.php(120): Magento\\Framework\\Reflection\\DataObjectProcessor->buildOutputDataArray(Object(Magento\\Sales\\Model\\Order\\Payment\\Interceptor), '\\\\Magento\\\\Sales\\\\...')\n#3 /srv/weldioc/html/vendor/magento/framework/Reflection/DataObjectProcessor.php(137): Magento\\Framework\\Reflection\\DataObjectProcessor->buildOutputDataArray(Object(Magento\\Sales\\Model\\Order), '\\\\Magento\\\\Sales\\\\...')\n#4 /srv/weldioc/html/vendor/magento/framework/Webapi/ServiceOutputProcessor.php(109): Magento\\Framework\\Reflection\\DataObjectProcessor->buildOutputDataArray(Object(Magento\\Sales\\Model\\ResourceModel\\Order\\Collection), '\\\\Magento\\\\Sales\\\\...')\n#5 /srv/weldioc/html/vendor/magento/framework/Webapi/ServiceOutputProcessor.php(60): Magento\\Framework\\Webapi\\ServiceOutputProcessor->convertValue(Object(Magento\\Sales\\Model\\ResourceModel\\Order\\Collection), '\\\\Magento\\\\Sales\\\\...')\n#6 /srv/weldioc/html/vendor/magento/module-webapi/Controller/Rest.php(312): Magento\\Framework\\Webapi\\ServiceOutputProcessor->process(Object(Magento\\Sales\\Model\\ResourceModel\\Order\\Collection), 'Magento\\\\Sales\\\\A...', 'getList')\n#7 /srv/weldioc/html/vendor/magento/module-webapi/Controller/Rest.php(216): Magento\\Webapi\\Controller\\Rest->processApiRequest()\n#8 /srv/weldioc/html/var/generation/Magento/Webapi/Controller/Rest/Interceptor.php(24): Magento\\Webapi\\Controller\\Rest->dispatch(Object(Magento\\Framework\\App\\Request\\Http))\n#9 /srv/weldioc/html/vendor/magento/framework/App/Http.php(135): Magento\\Webapi\\Controller\\Rest\\Interceptor->dispatch(Object(Magento\\Framework\\App\\Request\\Http))\n#10 /srv/weldioc/html/vendor/magento/framework/App/Bootstrap.php(258): Magento\\Framework\\App\\Http->launch()\n#11 /srv/weldioc/html/pub/index.php(46): Magento\\Framework\\App\\Bootstrap->run(Object(Magento\\Framework\\App\\Http))\n#12 {main}"
}
```

**Comment by YevSent** [comment]
Hi, @amnotafraid, this is known issue (MAGETWO-65144). Thanks for reporting.

**Comment by amnotafraid** [comment]
@joni-jones Thank you.  I searched for the issue, but I couldn't find it.  Would you please provide a link to MAGETWO-65144?

**Comment by YevSent** [comment]
Hi @amnotafraid, it's internal ticket.

**Comment by magento-engcom-team** [comment]
@amnotafraid, thank you for your report.
The issue is already fixed in 2.2.0
